crates/orderbook/src/orderbook.rs

```diff
@@ -239,22 +222,13 @@ impl Orderbook {
         Ok(solvable_orders)
     }
 
-    pub async fn get_auction(&self) -> Result<Auction> {
-        let solvable_orders = self.get_solvable_orders().await?;
-
-        let order_count = solvable_orders.orders.len();
-        let (orders, prices) =
-            get_orders_with_native_prices(solvable_orders.orders, &*self.native_price_estimator)
-                .await;
-        let filtered_orders = order_count - orders.len();
-        self.metrics.filtered_solvable_orders(filtered_orders);
-
-        Ok(Auction {
-            block: solvable_orders.block,
-            latest_settlement_block: solvable_orders.latest_settlement_block,
-            orders,
-            prices,
-        })
+    pub fn get_auction(&self) -> Result<Auction> {
+        let (auction, update_time) = self.solvable_orders.cached_auction();
+        ensure!(
+            update_time.elapsed() <= self.solvable_orders_max_update_age,
```

Comment:
Unrelated to this PR, but would it make sense to instead of checking whether or not it is out of date based on time, if it is out of date based on block numbers? I.e. you return the "last updated block" with the auction and compare it to the current block from the `CurrentBlockStream`? There are in rare cases 1min between blocks - and if we are at the end of this window but 30s late, it doesn't really matter because the state of the blockchain hasn't changed.

Also, slightly unrelated to this PR, could we add a historgram metric to get a better sense of how much and how often the auction is out of date? It would help to evaluate the impact of these changes

Reply:
Makes sense. In that case measuring orders that get left out makes more sense than auction delay.

Also, sorry if it wasn't clear before - this discussion wasn't blocking the merge for me (hence the approve), just thinking about how we can measure the impact of this change to get a better understanding at what the tradeoffs would be, and be able to pick the right constants for the time limits.